### ISSUE TITLE
Handle malformed login response

### ISSUE THREAD

**Issue by erorus** [favll/pogom#41]
As of commit eed1f968:
```
2016-07-27 13:13:33,011 [     search] [   INFO]  6765 reqs in 01m26s  [ 78.2 reqs/s 132.4KB/s (up)  92.7KB/s (down)] 0.09s av. per request
2016-07-27 13:13:34,798 [     search] [   INFO]  6790 reqs in 01m28s  [ 76.9 reqs/s 130.3KB/s (up)  91.2KB/s (down)] 0.09s av. per request
2016-07-27 13:13:34,799 [     search] [   INFO] Scan finished. Sleeping 91.70 seconds before continuing.
2016-07-27 13:15:06,519 [     search] [   INFO] Starting scan of 6790 locations
2016-07-27 13:15:06,520 [     search] [   INFO] Login has or is about to expire
2016-07-27 13:15:06,520 [     search] [   INFO] Attempting login
2016-07-27 13:15:06,527 [   auth_ptc] [   INFO] Login for: ***
2016-07-27 13:15:06,742 [   auth_ptc] [  ERROR] Could not decode response: <html><body>Not found (error to origin)</body></html>
2016-07-27 13:15:06,743 [     search] [   INFO] Login failed, retrying in 1.00 seconds
2016-07-27 13:15:07,749 [   auth_ptc] [   INFO] Login for: ***
2016-07-27 13:15:17,885 [   auth_ptc] [  ERROR] Server timed out
2016-07-27 13:15:17,887 [     search] [   INFO] Login failed, retrying in 1.80 seconds
2016-07-27 13:15:19,691 [   auth_ptc] [   INFO] Login for: ***
2016-07-27 13:15:29,872 [   auth_ptc] [  ERROR] Server timed out
2016-07-27 13:15:29,873 [     search] [   INFO] Login failed, retrying in 3.24 seconds
2016-07-27 13:15:33,120 [   auth_ptc] [   INFO] Login for: ***
2016-07-27 13:15:43,281 [   auth_ptc] [  ERROR] Server timed out
2016-07-27 13:15:43,284 [     search] [   INFO] Login failed, retrying in 5.84 seconds
2016-07-27 13:15:49,131 [   auth_ptc] [   INFO] Login for: ***
2016-07-27 13:15:59,346 [   auth_ptc] [  ERROR] Server timed out
2016-07-27 13:15:59,349 [     search] [   INFO] Login failed, retrying in 10.52 seconds
2016-07-27 13:16:09,877 [   auth_ptc] [   INFO] Login for: ***
2016-07-27 13:16:20,040 [   auth_ptc] [  ERROR] Server timed out
2016-07-27 13:16:20,041 [     search] [   INFO] Login failed, retrying in 18.94 seconds
2016-07-27 13:16:38,999 [   auth_ptc] [   INFO] Login for: ***
2016-07-27 13:16:49,204 [   auth_ptc] [  ERROR] Server timed out
2016-07-27 13:16:49,207 [     search] [   INFO] Login failed, retrying in 34.10 seconds
2016-07-27 13:17:23,336 [   auth_ptc] [   INFO] Login for: ***
2016-07-27 13:17:23,480 [   auth_ptc] [  ERROR] Could not decode response: <html><body>Not found (error to origin)</body></html>
2016-07-27 13:17:23,480 [     search] [   INFO] Login failed, retrying in 61.41 seconds
2016-07-27 13:18:24,939 [   auth_ptc] [   INFO] Login for: ***
2016-07-27 13:18:35,107 [   auth_ptc] [  ERROR] Server timed out
2016-07-27 13:18:35,108 [     search] [   INFO] Login failed, retrying in 110.60 seconds
2016-07-27 13:20:25,805 [   auth_ptc] [   INFO] Login for: ***
2016-07-27 13:20:36,000 [   auth_ptc] [  ERROR] Server timed out
2016-07-27 13:20:36,002 [     search] [   INFO] Login failed, retrying in 199.16 seconds
2016-07-27 13:23:55,232 [   auth_ptc] [   INFO] Login for: ***
2016-07-27 13:24:05,431 [   auth_ptc] [  ERROR] Server timed out
2016-07-27 13:24:05,432 [     search] [   INFO] Login failed, retrying in 300.00 seconds
2016-07-27 13:29:05,502 [   auth_ptc] [   INFO] Login for: ***
Exception in thread search_thread:
Traceback (most recent call last):
  File "/usr/local/lib/python2.7/threading.py", line 810, in __bootstrap_inner
    self.run()
  File "/usr/local/lib/python2.7/threading.py", line 763, in run
    self.__target(*self.__args, **self.__kwargs)
  File "pogom/pogom/search.py", line 233, in search_loop_async
    search_async(args)
  File "pogom/pogom/search.py", line 158, in search_async
    login_if_necessary(args, step_location)
  File "pogom/pogom/search.py", line 110, in login_if_necessary
    login(args, position)
  File "pogom/pogom/search.py", line 93, in login
    while not api.login(args.auth_service, args.username, args.password):
  File "pogom/pogom/pgoapi/pgoapi.py", line 151, in login
    if not self._rpc.auth_provider.login(username, password):
  File "pogom/pogom/pgoapi/auth_ptc.py", line 62, in login
    self.log.error('Could not decode response: {}'.format(r.content.decode('utf-8')))
UnicodeEncodeError: 'ascii' codec can't encode character u'\xe9' in position 296: ordinal not in range(128)
```

**Comment by erorus** [comment]
Actually I suppose this is a bug in pgoapi, but it might be nice if you caught it and treated it as a login failure and continue retrying.

Closing.

**Comment by nborrmann** [comment]
Fixed in 5c3b1bbc819097452f2ddc279bb16dbbc51000d9
